src/srv.rs

```diff
@@ -103,15 +109,19 @@ impl SrvResolver {
             }
 
             if !&hostname_parts[1..].ends_with(domain_name) {
-                srv_addresses.push(Err(ErrorKind::DnsResolve {
-                    message: format!(
-                        "SRV lookup for {} returned result {}, which does not match domain name {}",
-                        original_hostname,
-                        address,
-                        domain_name.join(".")
-                    ),
+                if matches!(dm, DomainMismatch::Error) {
```

Comment:
Based on the language in the spec (see the language in the second bullet [here](https://github.com/mongodb/specifications/blob/master/source/polling-srv-records-for-mongos-discovery/polling-srv-records-for-mongos-discovery.rst#implementation)), it seems like we should never actually return an error here and instead log a warning. I think these errors were just being constructed for the sake of logging them at some point in the future? But I think it would be simpler to do the logging here directly rather than propagate the error. Based on that I think we can just remove this logic entirely and file a ticket to log a warning when this happens.